### ISSUE TITLE
Set time expiration for isPastDue jobs

### ISSUE THREAD

**Issue by seandkim** [Azure/Azure-Functions#2242]
### TLDR
I want to reliably run my function with UseMonitor=true, but I do not want to run isPastDue jobs if the originalSchedule is more than x hour ago. 

### Detail
I have Azure function enabled on the "main" slot and disabled on one deployment slot. I deploy code to the deployment slot then swap slots. Whenever I do this, I see a job scheduled from several days ago get executed, probably because the suppressed job is enabled again. For example, the below image shows that a job that was originally scheduled from 7/7 got executed on 7/18. 

<img width="926" alt="Screen Shot 2022-07-18 at 4 53 05 PM" src="https://user-images.githubusercontent.com/20718126/179467393-2b46620e-c0ff-40b0-84e1-42fa4e1709b3.png">

[This doc](https://feedback.azure.com/forums/355860-azure-functions) states that I can disable pastDue jobs completely by setting `UseMonitor=false`. But I do not want to disable all pastDue jobs because I still want my job to reliably run despite Azure function downtime. At the same time, I don't want a job from several days ago to randomly get executed. 

It would be nice if we can specify time expiration for isPastDue jobs e.g. execute isPastDue jobs only if it is less than x minutes/hours/days ago.

**Comment by umangsriv** [comment]
@seandkim Please let me know if the issue still exists.

**Comment by umangsriv** [comment]
Closing the issue as there is no recent activity.
